dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.ChatCompletion.cs

```diff
@@ -190,151 +177,48 @@ internal async Task<IReadOnlyList<ChatMessageContent>> GetChatMessageContentsAsy
                 activity?.SetCompletionResponse([chatMessageContent], chatCompletion.Usage.InputTokens, chatCompletion.Usage.OutputTokens);
             }
 
-            // If we don't want to attempt to invoke any functions, just return the result.
-            if (!toolCallingConfig.AutoInvoke)
+            // If we don't want to attempt to invoke any functions or there is nothing to call, just return the result.
+            if (!functionCallingConfig.AutoInvoke || chatCompletion.ToolCalls.Count == 0)
             {
                 return [chatMessageContent];
             }
 
-            // Get our single result and extract the function call information. If this isn't a function call, or if it is
-            // but we're unable to find the function or extract the relevant information, just return the single result.
-            // Note that we don't check the FinishReason and instead check whether there are any tool calls, as the service
-            // may return a FinishReason of "stop" even if there are tool calls to be made, in particular if a required tool
-            // is specified.
-            if (chatCompletion.ToolCalls.Count == 0)
-            {
-                return [chatMessageContent];
-            }
-
-            if (this.Logger.IsEnabled(LogLevel.Debug))
-            {
-                this.Logger.LogDebug("Tool requests: {Requests}", chatCompletion.ToolCalls.Count);
-            }
-            if (this.Logger.IsEnabled(LogLevel.Trace))
+            // Process function calls by invoking the functions and adding the results to the chat history.
+            // Each function call will trigger auto-function-invocation filters, which can terminate the process.
+            // In such cases, we'll return the last message in the chat history.
+#pragma warning disable CS0618 // Type or member is obsolete
```

Comment:
Why is new code using something that is obsolete?

Reply:
The `ToolCallBehavior.ToolCallResultSerializerOptions` belong to the current/old model that we need to support until we decide to deprecate it. The new model does not have it, and null is passed as value here.

Reply:
or your idea is not supply it as argument at all because it's obsolete?

Reply:
removed